### PR TITLE
feat: forward platform-type to module operator deployments

### PR DESCRIPTION
**Pull request by tsisodia10**
## Summary

Forward the platform type (e.g. `XKS`, `OpenDataHub`, `SelfManagedRHOAI`) from the main operator to module operator Deployments via the `ODH_MODULE_OPERATOR_PLATFORM_TYPE` environment variable.

Module operators like ai-gateway-operator need the platform type to select platform-specific manifests (e.g. xKS overlays that exclude OpenShift-specific monitoring and service-serving cert annotations). Currently, module operators have no way to know the platform type because `injectModuleEnv` only forwards `APPLICATIONS_NAMESPACE`, `MONITORING_NAMESPACE`, and `RELATED_IMAGE_*` variables.

## Changes

- Add `PlatformType` field to `ModuleEnvInjection` struct (`pkg/controller/types/types.go`)
- Populate `PlatformType` from `platformCtx.Release.Name` in `provisionModules` (`modules_controller_actions.go`)
- Inject `ODH_MODULE_OPERATOR_PLATFORM_TYPE` env var into module Deployments, following the same pattern as `APPLICATIONS_NAMESPACE` and `MONITORING_NAMESPACE` (`modules_controller_actions_inject_env.go`)
- Add tests for both set and empty platform-type cases

## Context

This is a companion to [ai-gateway-operator#58](https://github.com/opendatahub-io/ai-gateway-operator/pull/58) which adds xKS overlay selection for MaaS in ai-gateway-operator. Without this PR, `cfg.PlatformType` in ai-gateway-operator defaults to `"unknown"` and the xKS overlay is never selected.

Follows zdtsw's guidance that platform identity should be passed down from the operator rather than auto-detected by each module.

Related: RHAISTRAT-1377

## Test plan

- [x] `go test ./internal/controller/modules/ -run TestInjectModuleEnv` — all 10 tests pass
- [x] New `TestInjectModuleEnvPlatformType` verifies XKS value is injected
- [x] New `TestInjectModuleEnvEmptyPlatformType` verifies empty string is not injected

## Summary by CodeRabbit

- **New Features**
  - Module-managed deployments now receive a platform type environment variable when configured.
  - The platform type is automatically populated from the release configuration and forwarded to module operators.

- **Bug Fixes**
  - Platform type injection is applied consistently; empty platform values are omitted without impacting other injected environment variables.

- **Tests**
  - Added/updated tests for both populated and empty platform type scenarios.

- **Documentation**
  - Updated `injectModuleEnv` documentation to reflect the additional platform identity variables.

### CODE REVIEW

**Review by davidebianchi — COMMENTED** [review]
lgtm

@StevenTobin can you take a look? It's probably better to forward that information, instead of have each module to recompute it to avoid incongruences

**Comment by StevenTobin** [comment]
> It's probably better to forward that information

Yep this makes sense. The only Caveat is that I think `release.Name` returns the human readable name, on XKS this is fine but consumers would need to be aware that they wouldn't get `OpenDataHub` or `SelfManagedRHOAI` they would get `Open Data Hub` and `OpenShift AI Self-Managed`

/lgtm

**Comment by davidebianchi** [comment]
> Yep this makes sense. The only Caveat is that I think release.Name returns the human readable name, on XKS this is fine but consumers would need to be aware that they wouldn't get OpenDataHub or SelfManagedRHOAI they would get Open Data Hub and OpenShift AI Self-Managed

@StevenTobin yes, it returns the readable name. Constants are already available in the operator utilities: https://github.com/opendatahub-io/odh-platform-utilities/blob/a8ae1b749412e9d68e6d15c1974ed533593355fb/pkg/cluster/types.go#L13-L27, which can be used by module team to check it.

File: pkg/controller/types/types.go [hunk 1]
```diff
@@ -37,6 +38,10 @@ type ModuleEnvInjection struct {
 	// MonitoringNamespace is the platform's monitoring namespace. "" when
 	// monitoring is not configured or DSCI not exist (e.g xks).
 	MonitoringNamespace string
+	// PlatformType is the platform identifier (e.g. "OpenDataHub",
+	// "SelfManagedRHOAI", "XKS"). Forwarded to module operators so they
+	// can select platform-specific manifests without auto-detecting.
+	PlatformType string
```
Comment (davidebianchi) [inline pkg/controller/types/types.go]: The used values are the readable one, as defined [here](https://github.com/opendatahub-io/opendatahub-operator/blob/main/pkg/cluster/const.go#L6-L15), correct? 
It could also be set as common.Platform instead of string, to be more clear about the possible values?

Comment (tsisodia10) [reply pkg/controller/types/types.go]: Good suggestion, updated to common.Platform in 706ece3. The field is now typed, making the valid values (OpenDataHub, SelfManagedRHOAI, ManagedRHOAI, XKS) explicit.

**Review by davidebianchi — APPROVED** [review]

**Review by davidebianchi — APPROVED** [review]

**Review by zdtsw — APPROVED** [review]
